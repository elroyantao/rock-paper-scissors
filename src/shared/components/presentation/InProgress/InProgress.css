.InProgress {
  display: flex;
  justify-content: center;
  align-items: center;
}

.InProgress-player {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.InProgress-label {

}
