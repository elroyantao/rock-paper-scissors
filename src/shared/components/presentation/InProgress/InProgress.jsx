import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Hand from '../Hands/Hand'
import styles from './InProgress.css'

@connect((state) => ({
  playerHand: state.hand.playerHand,
  computerHand: state.hand.computerHand,
  level: state.level.name
}))
export default class InProgress extends Component {
  static propTypes = {
    playerHand: PropTypes.number,
    computerHand: PropTypes.number,
    level: PropTypes.string
  }

  render() {
    const { playerHand, computerHand, level } = this.props
    if (playerHand === null) return null
    const computerType = computerHand === null ? `${level}Waiting` : computerHand

    return (
      <div className={styles.InProgress}>
        <div className={styles['InProgress-player']}>
          <h3 className={styles['InProgress-label']}>You choose</h3>
          <Hand type={playerHand} isProgress />
        </div>
        <div className={styles['InProgress-player']}>
          <h3 className={styles['InProgress-label']}>computer choose</h3>
          <Hand type={computerType} isProgress />
        </div>
      </div>
    )
  }
}
