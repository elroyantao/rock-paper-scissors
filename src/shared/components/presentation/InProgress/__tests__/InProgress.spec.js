import testComponentHelper from '../../../../lib/test-helper'
import InProgress from '../InProgress'


describe('<InProgress /> ', () => {
  const initialState = {
    playerHand: null,
    computerHand: null,
    level: 'classic'
  }
  const renderComponent = testComponentHelper(InProgress.WrappedComponent)
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialState).getTree())
        .toMatchSnapshot()
    })
    it('when playerHand is not null and computerHand is null', () => {
      expect(renderComponent({
        ...initialState,
        playerHand: 3
      }).getTree())
        .toMatchSnapshot()
    })
    it('when playerHand and computerHand is not null', () => {
      expect(renderComponent({
        ...initialState,
        playerHand: 3,
        computerHand: 2
      }).getTree())
        .toMatchSnapshot()
    })
    it('when playerHand and computerHand is not null and level is advanced', () => {
      expect(renderComponent({
        ...initialState,
        playerHand: 3,
        level: 'advanced'
      }).getTree())
        .toMatchSnapshot()
    })
  })
})
