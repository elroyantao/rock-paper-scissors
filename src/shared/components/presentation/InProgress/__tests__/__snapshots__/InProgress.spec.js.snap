// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InProgress />  @renders in default state 1`] = `null`;

exports[`<InProgress />  @renders when playerHand and computerHand is not null 1`] = `
<div
  className="InProgress"
>
  <div
    className="InProgress-player"
  >
    <h3
      className="InProgress-label"
    >
      You choose
    </h3>
    <Hand
      isProgress={true}
      onClick={[Function]}
      type={3}
    />
  </div>
  <div
    className="InProgress-player"
  >
    <h3
      className="InProgress-label"
    >
      computer choose
    </h3>
    <Hand
      isProgress={true}
      onClick={[Function]}
      type={2}
    />
  </div>
</div>
`;

exports[`<InProgress />  @renders when playerHand and computerHand is not null and level is advanced 1`] = `
<div
  className="InProgress"
>
  <div
    className="InProgress-player"
  >
    <h3
      className="InProgress-label"
    >
      You choose
    </h3>
    <Hand
      isProgress={true}
      onClick={[Function]}
      type={3}
    />
  </div>
  <div
    className="InProgress-player"
  >
    <h3
      className="InProgress-label"
    >
      computer choose
    </h3>
    <Hand
      isProgress={true}
      onClick={[Function]}
      type="advancedWaiting"
    />
  </div>
</div>
`;

exports[`<InProgress />  @renders when playerHand is not null and computerHand is null 1`] = `
<div
  className="InProgress"
>
  <div
    className="InProgress-player"
  >
    <h3
      className="InProgress-label"
    >
      You choose
    </h3>
    <Hand
      isProgress={true}
      onClick={[Function]}
      type={3}
    />
  </div>
  <div
    className="InProgress-player"
  >
    <h3
      className="InProgress-label"
    >
      computer choose
    </h3>
    <Hand
      isProgress={true}
      onClick={[Function]}
      type="classicWaiting"
    />
  </div>
</div>
`;
